File: lab3/FileReverse.c
//-----------------------------------------------------------------------------
// FileIO.c
// Reads input file and prints each word on a separate line of
// the output file.
// Rhea Lingaiah
// rlingaia@ucsc.edu
// lab3 12b
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

void stringReverse(char* s){
	int i=0;
	int j= strlen(s)-1;
	while(i<=j){
		int a= s[i];
		int b= s[j];
		s[i]=b;
		s[j]=a;
		i++;
		j--;
	}
}
int main(int argc, char * argv[]){
   FILE *in, *out;   // handles for input and output files 
   char word[256];   // char array to store words from input file 

   // check command line for correct number of arguments
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   // open input file for reading 
   in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   // open output file for writing 
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   // read words from input file, print on separate lines to output file 
   while(fscanf(in, " %s", word)!=EOF){
   		stringReverse(word);
      fprintf(out, "%s\n", word);
   }

   // close input and output files 
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);
}

File: lab2/FileReverse.java
/**
 * Created by Rhea lingaiah on 7/9/17.
 * 12B
 * rlingaia@ucsc.edu
 */

import java.io.*;
import java.util.Scanner;

public class FileReverse {
    public static void main(String[] args) throws IOException {

        Scanner in = null;
        PrintWriter out = null;
        String line = null;
        String[] reverse = null;
        int n, lineNumber = 0;

        // check number of command line arguments is at least 2
        if (args.length < 2) {
            System.out.println("Usage: FileReverse <input file> <output file>");
            System.exit(1);
        }

        // open files
        in = new Scanner(new File(args[0]));
        out = new PrintWriter(new FileWriter(args[1]));

        // read lines from in, extract and print tokens from each line
        while (in.hasNextLine()) {
            lineNumber++;

            // trim leading and trailing spaces, then add one trailing space so
            // split works on blank lines
            line = in.nextLine().trim() + " ";

            // split line around white space
            reverse = line.split("\\s+");
            for(int i=0;i<reverse.length;i++){
                String stringReversed= stringReverse(reverse[i], reverse[i].length());
                out.println(stringReversed);
            }
        }

        // close files
        in.close();
        out.close();
    }

    public static String stringReverse(String s, int n){
        if (s.length()<2) {
            return s;
        }
        else
            return stringReverse(s.substring(1), n)+ s.charAt(0);

    }
}
